internal/api/block_schemas.go

```diff
@@ -0,0 +1,38 @@
+package api
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+// BlockSchemaClient is a client for working with block schemas.
+type BlockSchemaClient interface {
+	List(ctx context.Context, blockTypeIDs []uuid.UUIDs) ([]*BlockSchema, error)
+}
+
+// BlockSchema is a representation of a block schema.
+type BlockSchema struct {
+	BaseModel
+	BlockType
```

Comment:
im a little hazy with embeddings <> json serialization, but if the API payload looks something like:

```json
{
  "id": "06c34bb7-ac74-4dbb-bcdb-407daa305e74",
  "created": "2024-04-05T19:55:34.265851+00:00",
  "updated": "2024-04-05T19:55:34.265867+00:00",
  "checksum": "sha256:273dfe712133cba3cdb092d88741da37e33cee892bfcf4fcf2895bdcee49a965",
  "fields": {...},
  "block_type_id": "063573f9-e491-48a4-8b2e-c1905a890c7e",
  "block_type": {
    "id": "063573f9-e491-48a4-8b2e-c1905a890c7e",
    "created": "2023-11-09T20:32:13.680977+00:00",
    "updated": "2023-11-09T20:32:13.681003+00:00",
    "name": "Slack Webhook",
    "slug": "slack-webhook",
    "logo_url": "https://cdn.sanity.io/images/3ugk85nk/production/c1965ecbf8704ee1ea20d77786de9a41ce1087d1-500x500.png",
    "documentation_url": "https://docs.prefect.io/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.SlackWebhook",
    "description": "Enables sending notifications via a provided Slack webhook.",
    "code_example": "Load a saved Slack webhook and send a message:\n```python\nfrom prefect.blocks.notifications import SlackWebhook\n\nslack_webhook_block = SlackWebhook.load(\"BLOCK_NAME\")\nslack_webhook_block.notify(\"Hello from Prefect!\")\n```",
    "is_protected": false
  },
  "capabilities": [
    "notify"
  ],
  "version": "2.16.9"
}
```

should we be setting a `BlockType` property here instead of embedding?  

```golang
type BlockSchema struct {
	BaseModel
	BlockType    BlockType   `json:"block_type"`
        BlockTypeID uuid.UUID   `json:"block_type_id"`
...
}
```

Reply:
Potentially, although I figured this would work because we're already using that approach with `BaseModel`:

```go
// internal/api/base_model.go

// BaseModel is embedded in all other types and defines fields
// common to all Prefect data models.
type BaseModel struct {
	ID      uuid.UUID  `json:"id"`
	Created *time.Time `json:"created"`
	Updated *time.Time `json:"updated"`
}
```

🤔

Reply:
I see, I think you're right because embeddeding doesn't put a struct inside of it - it just inherits the values of that type, which isn't what we want. Fixing.

Reply:
Updated in 65fbb37.

Reply:
right on